precompiles/werc20/tx.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package werc20
+
+const (
+	// DepositMethod defines the ABI method name for the IWERC20 deposit
+	// transaction.
+	DepositMethod = "deposit"
+	// WithdrawMethod defines the ABI method name for the IWERC20 withdraw
+	// transaction.
+	WithdrawMethod = "withdraw"
+)
+
+// Deposit is a no-op and mock function that provides the same interface as the
+// WETH contract to support equality between the native coin and its wrapped
+// ERC-20 (e.g. EVMOS and WEVMOS).
+func (p Precompile) Deposit() ([]byte, error) {
```

Comment:
Since this is a payable function, the contract will receive the coins and do nothing. It should send the coins back to the sender resulting in no balance change.